src/logging.rs

```diff
@@ -78,12 +78,49 @@ fn is_valid_field(input: &str) -> bool {
     true
 }
 
-fn add_field_and_payload(data: &mut String, field: &str, payload: &str) {
+/// Add `field` and `payload` to journal fields `data` with explicit length encoding.
+///
+/// Write
+///
+/// 1. the field name,
+/// 2. an ASCII newline,
+/// 3. the payload size as LE encoded 64 bit integer,
+/// 4. the payload, and
+/// 5. a final ASCII newline
+///
+/// to `data`.
+///
+/// See <https://systemd.io/JOURNAL_NATIVE_PROTOCOL/> for details.
+fn add_field_and_payload_explicit_length(data: &mut Vec<u8>, field: &str, payload: &str) {
```

Comment:
I've moved this to a dedicated function to be able to test this independently, and also because technically journald allows this encoding for all data, even if the payload doesn't contain newlines, so it can stand independently.